convert2rhel/subscription.py

```diff
@@ -179,7 +180,15 @@ def register_system():
         registration_cmd = RegistrationCommand.from_tool_opts(tool_opts)
 
         try:
+            # The file /etc/os-release is needed for subscribing the system and is being removed with
+            # <system-name>-release package in one of the steps before
+            # RHELC-16
+            os_release_file.restore(rollback=False)
             registration_cmd()
+            # Need to remove the file, if it would stay there would be leftover /etc/os-release.rpmorig
+            # after conversion
+            # RHELC-16
+            os_release_file.remove()
```

Comment:
This part feels a little unclean but we can fix it in the next release.  I've written up what's unclean about it and how we can fix it later here: https://issues.redhat.com/browse/RHELC-809